File: staircase.py
# Solving the Hacker Rank problem found here: https://www.hackerrank.com/challenges/staircase/problem?utm_campaign=challenge-recommendation&utm_medium=email&utm_source=7-day-campaign


def staircase(hashtags):
    '''Given an integer number of hashtags, return a right aligned staircase of
       those characters.

    '''
    quota = 1
    for i in range(hashtags):
        line = ""
        for j in range(hashtags):
            if hashtags - j <= quota:
                line += "#"
            else:
                line += " "
        print(line)
        quota += 1
    return None


if __name__ == "__main__":
    staircase(4)

File: staircase.java
import java.util.Scanner;


public class Staircase {
    //-----------------------------------------------------
    // Solution to the Hacker Rank problem found here:
    // Solving the Hacker Rank problem found below::
    // https://www.hackerrank.com/challenges/staircase/problem?utm_campaign=challenge-recommendation&utm_medium=email&utm_source=7-day-campaign
    //-----------------------------------------------------

    /**********************************************************************
     * printStaircase
     * pre: int steps is number of steps in the staircase
     * post: prints a right-aligned staircase with steps number of rows,
     *       and an increasing number of # symbols going down the rows
     *********************************************************************/
     public static void printStaircase(int steps)
     {

     }

     /*******************************************************************
     * Driver method
     * pre: None
     * post: invokes the printStaircase method once and passes in input
     *       from user
     *********************************************************************/
     public static void main(String[] args)
     {
         // take input from user on number of steps
         Scanner scan = new Scanner(System.in);
         System.out.print("Enter number of step in staircase: ");
         int steps = scan.nextInt();

         // printing the staircase
         System.out.println("------------------------------------------------");
         System.out.println("Here You Go! Have Fun and Reach Your Dreams!");
         printStaircase(steps);

     }
 }
